docker/Dockerfile

```diff
@@ -0,0 +1,81 @@
+# Copyright 2023 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Builder image
+###############################################################################
+FROM redhat/ubi9 AS builder
+
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETPLATFORM
+ARG KUBESTELLAR_VERSION=v0.3.1
```

Comment:
The latest is v0.4.0 now

Reply:
@francostellari to be safe I would change this to take an external var during the "docker build" so that no matter what we do in the future, you are protected from a hard reference.  The input to the docker build command can include a hardcoded value or a reference to a local env var.

either:
docker build --build-arg KUBESTELLAR_VERSION=0.3.1

or: 
docker build --build-arg KUBESTELLAR_VERSION=${KUBESTELLAR_VERSION}

and line 23 would be 
"ARG KUBESTELLAR_VERSION=KUBESTELLAR_VERSION"

Reply:
@clubanderson : the Makefile is already doing `--build-arg KUBESTELLAR_VERSION`.  So I do not think we need a default value inside the Dockefile at all.

More importantly, however, I do not think this Dockerfile should be fetching KubeStellar.  Building this image should be like building anything else in KubeStellar.  It works with whatever the contributor already has locally, whether that was a release, a branch, or a work-in-progress not yet pushed.

Reply:
That's fair too. Just use what is local. That's another good option.

Reply:
Remember, as long as the Dockerfile insists on fetching a release, a contributor can not test without making a release.  That is backwards.

Reply:
@MikeSpreitzer @clubanderson I have removed the default value from the `Dockerfile`